pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -240,9 +249,21 @@ func (p *staticPolicy) RemoveContainer(s state.State, containerID string) (rerr
 	return nil
 }
 
-func (p *staticPolicy) allocateCPUs(s state.State, numCPUs int) (cpuset.CPUSet, error) {
-	klog.Infof("[cpumanager] allocateCpus: (numCPUs: %d)", numCPUs)
-	result, err := takeByTopology(p.topology, p.assignableCPUs(s), numCPUs)
+func (p *staticPolicy) allocateCPUs(s state.State, numCPUs int, socketmask socketmask.SocketMask) (cpuset.CPUSet, error) {
+	klog.Infof("[cpumanager] allocateCpus: (numCPUs: %d, socket: %v)", numCPUs, socketmask)
+	assignableCPUs := cpuset.NewCPUSet()
+	var sockets []int
+	if socketmask != nil {
+		sockets = socketmask.GetSockets()
+	}
+	if len(sockets) != 0 {
+		for _, socketID := range sockets {
+			assignableCPUs = assignableCPUs.Union(p.assignableCPUs(s).Intersection(p.topology.CPUDetails.CPUsInSocket(socketID)))
+		}
+	} else {
+		assignableCPUs = p.assignableCPUs(s)
+	}
+	result, err := takeByTopology(p.topology, assignableCPUs, numCPUs)
```

Comment:
This logic is currently buggy. It will  only pull CPUs from aligned sockets, regardless of the `TopologyManager` in use. Per offiline discussions, we will go ahead with merging this PR with this bug in place.

I will create an issue explaining the bug in more detail, with a follow-on PR (including some more cleanup) before the release.

Reply:
Fixed in latest revision of this PR.